tutorial/numpy-functions.md

```diff
@@ -0,0 +1,385 @@
+      4_C_INT64_T, 2_C_INT64_T])
+    allocate(nrm2(0:1_C_INT64_T))
+    nrm2 = Norm2(Real(arr2, C_DOUBLE),1_C_INT64_T)
+    print *, nrm
+
+    end program prog_boo
+    ```
+
+## [Real](https://numpy.org/doc/stable/reference/generated/numpy.real.html) and [imag](https://numpy.org/doc/stable/reference/generated/numpy.imag.html) functions
+
+-   Supported languages: C, fortran
+
+-   python code:
+
+    ```python
+    from numpy import imag, real, array
+    arr1 = array([1+1j,2+1j,3+1j,4+1j])
+    real_part = real(arr1)
+    imag_part = imag(arr1)
+    print("real part for arr1: " , real_part, "\nimag part for arr1: ", imag_part)
+    ```
+
+-   fortran equivalent:
+
+    ```fortran
+    program prog_boo
+
+    use, intrinsic :: ISO_C_BINDING
+
+    implicit none
+
+    complex(C_DOUBLE_COMPLEX), allocatable :: arr1(:)
+    real(C_DOUBLE), allocatable :: real_part(:)
+    real(C_DOUBLE), allocatable :: imag_part(:)
+
+    allocate(arr1(0:3_C_INT64_T))
+    arr1 = [(1.0_C_DOUBLE, 1.0_C_DOUBLE), (2.0_C_DOUBLE, 1.0_C_DOUBLE), ( &
+          3.0_C_DOUBLE, 1.0_C_DOUBLE), (4.0_C_DOUBLE, 1.0_C_DOUBLE)]
+    allocate(real_part(0:3_C_INT64_T))
+    real_part = Real(arr1, C_DOUBLE)
+    allocate(imag_part(0:3_C_INT64_T))
+    imag_part = aimag(arr1)
+    print *, 'real part for arr1: ' // ' ' , real_part, ACHAR(10) // 'imag part for arr1: ' // ' ' , imag_part
+
+    end program prog_boo
+    ```
+
+-   C equivalent:
+
+    ```C
+    #include <stdint.h>
+    #include <stdlib.h>
+    #include <stdio.h>
+    #include "ndarrays.h"
+    #include <complex.h>
+    int main()
+    {
+        t_ndarray arr1;
+        t_ndarray real_part;
+        t_ndarray imag_part;
+        int64_t i_0001;
+        int64_t i;
+        int64_t i_0002;
+        arr1 = array_create(1, (int64_t[]){4}, nd_cdouble);
+        double complex array_dummy_0001[] = {(1.0 + 1.0 * _Complex_I), (2.0 + 1.0 * _Complex_I), (3.0 + 1.0 * _Complex_I), (4.0 + 1.0 * _Complex_I)};
+        memcpy(arr1.nd_cdouble, array_dummy_0001, arr1.buffer_size);
+        real_part = array_create(1, (int64_t[]){4}, nd_double);
+        for (i_0001 = 0; i_0001 < 4; i_0001 += 1)
+        {
+            real_part.nd_double[get_index(real_part, i_0001)] = creal(arr1.nd_cdouble[get_index(arr1, i_0001)]);
+        }
+        imag_part = array_create(1, (int64_t[]){4}, nd_double);
+        for (i_0001 = 0; i_0001 < 4; i_0001 += 1)
+        {
+            imag_part.nd_double[get_index(imag_part, i_0001)] = cimag(arr1.nd_cdouble[get_index(arr1, i_0001)]);
+        }
+        printf("%s ", "real part for arr1: ");
+        printf("%s", "[");
+        for (i = 0; i < 4 - 1; i += 1)
+        {
+            printf("%.12lf ", real_part.nd_double[get_index(real_part, i)]);
+        }
+        printf("%.12lf]", real_part.nd_double[get_index(real_part, 4 - 1)]);
+        printf("%s ", "\nimag part for arr1: ");
+        printf("%s", "[");
+        for (i_0002 = 0; i_0002 < 4 - 1; i_0002 += 1)
```

Comment:
This could also be simplified to something neater. After a little investigation this is my fault. Could you please change [this line](https://github.com/pyccel/pyccel/blob/master/pyccel/codegen/printing/ccode.py#L763) to use `orig_args[i].shape[0]` instead of `PythonLen(orig_args[i])`